### ISSUE TITLE
Subscriptions: Strange Invoice Price resulting from #42221

### ISSUE THREAD

**Issue by david-loe** [frappe/erpnext#42779]
some how the price of the resulting invoices is higher/lower than it should be, depending on the distance of the subscription period to the current date. A strange margin is added.

Start Date: 16th June
Interval: 1 Month
Price of Subscriotion: 80€

Create 2 Invoices with new force action

Invoice#1: 154,67€
Invoice#2: 72,26€
![grafik](https://github.com/user-attachments/assets/deda6fed-66ce-44fc-984a-73d0c525a925)

_Originally posted by @david-loe in https://github.com/frappe/erpnext/issues/42221#issuecomment-2289662224_

**Comment by ruthra-kumar** [comment]
What type of subscription price is set? 

Could you share the Subscription plan details.

@david-loe

**Comment by david-loe** [comment]
![Screenshot_20240821-124354.png](https://github.com/user-attachments/assets/4cb048c8-ec8e-49f7-9254-8e1fb57ad785)

**Comment by ruthra-kumar** [comment]
Can't replicate.

With the similar Subscription plan setup and an Item Price of 100 /-, Invoices do get created with the Price list rate.
<img width="1089" alt="Screenshot 2024-08-21 at 5 34 11 PM" src="https://github.com/user-attachments/assets/4068e325-c2aa-4caf-85db-e15b05653538">

<img width="1049" alt="Screenshot 2024-08-21 at 5 32 28 PM" src="https://github.com/user-attachments/assets/9e5cf745-a7d9-45d4-a00c-62e16f45e020">
<img width="1150" alt="Screenshot 2024-08-21 at 5 32 45 PM" src="https://github.com/user-attachments/assets/06652def-d57e-484b-b460-9b70bdd65f27">

**Comment by ruthra-kumar** [comment]
Is the customer billed in company currency or a foreign currency?

**Comment by david-loe** [comment]
Strange.. Currency is always EUR

I just tried again and the same Error accours:
Start Date: 16th June
Interval: 1 Month

Create 2 Invoices with new force action

Invoice#1: 178,67€
Invoice#2: 95,48€


I'm using the following Versions:
ERPNext: v15.33.0
Frappe Framework: v15.38.0

**Comment by david-loe** [comment]
![grafik](https://github.com/user-attachments/assets/54b17341-c01a-4cc6-93fb-f6b4bc8c9086)

**Comment by david-loe** [comment]
@ruthra-kumar 
I found the difference:
When turning `Generate Invoice At` to `Beginning of the current subscription period` everything is fine.
But when using `End of current subscription period` the wrong price accours

**Comment by ruthra-kumar** [comment]
> @ruthra-kumar I found the difference: When turning `Generate Invoice At` to `Beginning of the current subscription period` everything is fine. But when using `End of current subscription period` the wrong price accours

Yup. It's replicable when `End of current subscription period` is set.

**Comment by ruthra-kumar** [comment]
@david-loe 
This is caused by 'Prorate' option in 'Subscription Settings'. If you want the price list rate, disable it.

**Comment by david-loe** [comment]
Okay

But this calculation then ist false. The resulting price when using prorate should never exceed 100% of the default pirce.
This may never accoured before as one was not able to generate subscribtion invoices from the past, but with the new force action the calculation has to be revisited I think.
